### ISSUE TITLE
Consolidate input environment variables used to collect required and optional metadata for the agent

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#2203]
Wow; lots of these, with namespace all over the place. This is definitely something we should aspire to improve in the the future!

_Originally posted by @dbutenhof in https://github.com/distributed-system-analysis/pbench/pull/2181#r607056745_

**Comment by webbnh** [comment]
Yes, but...what kind of latitude do we have for making changes to them?  Aren't we almost completely restricted by binary and user compatibility?  (I.e., aren't nearly all of these documented, despite their variety of styles?)

**Comment by portante** [comment]
As we productize, we can declare a cut-over date so that we use all "pbench" or "PBENCH_*" environment variables, adding a shim later which would issue a warning, about being dropped in the 1.0 version.
